src/libs/structs/state.rs

```diff
@@ -71,7 +72,33 @@ pub struct StrataState<BackendData: Backend + 'static> {
 	pub workspaces: Workspaces,
 	pub pointer_location: Point<f64, Logical>,
 }
-
+//
+// pub struct GlobalState<BackendData: Backend + 'static> {
+// 	inner: OnceCell<StrataState<BackendData>>,
+// }
+//
+// impl<BackendData: Backend + 'static> GlobalState<BackendData> {
+// 	pub fn new() -> Self {
+// 		Self { inner: OnceCell::new() }
+// 	}
+//
+// 	pub fn set(&self, state: StrataState<BackendData>) -> Result<(), String> {
+// 		self.inner.set(state).map_err(|_| "Failed to set StrataState in GlobalState".to_owned())
+// 	}
+//
+// 	pub fn get(&self) -> Mutex<StrataState<BackendData>> {
+// 		self.inner.get()
+// 	}
+// }
+//
+// impl<BackendData: Backend + 'static> Deref for GlobalState<BackendData> {
+// 	type Target = StrataState<BackendData>;
+//
+// 	fn deref(&self) -> Self::Target {
+// 		self.get().expect("Uninitialized")
+// 	}
+// }
```

Comment:
```suggestion
// pub struct GlobalState {
// 	inner: OnceCell<StrataState>,
// }
//
// impl GlobalState {
// 	pub fn new() -> Self {
// 		Self { inner: OnceCell::new() }
// 	}
//
// 	pub fn set(&self, state: StrataState) -> Result<(), String> {
// 		self.inner.set(state).map_err(|_| "Failed to set StrataState in GlobalState".to_owned())
// 	}
//
// 	pub fn get(&self) -> &StrataState {
// 		self.inner.get()
// 	}
// }
//
// impl Deref for GlobalState {
// 	type Target = &StrataState;
//
// 	fn deref(&self) -> Self::Target {
// 		self.get().expect("Uninitialized")
// 	}
// }
```

Reply:
Okay! I will try this and get back to you. But currently I'm getting an error not related to this (or so I think):
```
error[E0609]: no field `backend` on type `Box<(dyn libs::structs::state::Backend + 'static)>`
```
In `state.rs`, I've changed:
```diff
pub fn new(
	mut loop_handle: LoopHandle<'static, CalloopData>,
	loop_signal: LoopSignal,
	display: &mut Display<StrataState>,
-       backend_data: BackendData,
+	backend_data: Box<dyn Backend>,
) 
```

You've said in one of your previous comments:

> Then you can just initialize it with `Box::new(MyBackend::new())` where `MyBackend` implements the `Backend` trait.

And I'm pretty sure this error occurs cause I haven't initialized it somewhere. Also does `MyBackend` here imply one of our backends such as `WinitData`? `WinitData` implements the `Backend` trait here:
```
impl Backend for WinitData {
	fn seat_name(&self) -> String {
		"winit".to_string()
	}
}
```

Could you please tell me where I would initialize the variable and if what I've understood is even right? Thanks :)

Reply:
Yep, `WinData` is what I meant. Basically you need to have it initialized whenever you initialize `StrataState`. What this means in our case is that the `StrataState::new()` function needs to take the backend as an argument (or an enum representing it so that it can create it itself).

I don't have access to a computer right now so I can't look into the code much but I'll get back to you with more details tomorrow.

Reply:
Okay, np :)